apache/superset
# fix(db_engine_specs): Update convert_dttm to work correctly with CrateDB

### SUMMARY
[By design](https://cratedb.com/docs/crate/reference/en/5.6/general/ddl/data-types.html#timestamp), when CrateDB casts `BIGINT` values to `TIMESTAMP` it interprets them as being in milliseconds but when it casts `DOUBLE` values it interprets them as being in seconds.
This means that when Superset was trying to filter on a value such as 1679184000000.0 this was interpreted as a date far in the future instead of 2024.

Fixes #27382

### TESTING INSTRUCTIONS

- Spin up a CrateDB instance
- Create a table with a `TIMESTAMP WITH TIME ZONE` column
- Spin up Superset with `docker-compose -f docker-compose-non-dev.yml up`
- Browse http://localhost:8088/ and login with admin / admin
- Create a database connection from Superset to CrateDB using `crate://username:password@hostname:4200/?ssl=true`
- Create a dataset pointing to the table created earlier
- Create a new table chart
- Define dimensions and metrics
- Add a filter on the timestamp column for last week or a custom interval of dates
- Check in CrateDB the query that was run with `SELECT * FROM sys.jobs_log ORDER BY started DESC;`
- Edit `/app/superset/db_engine_specs/crate.py` as in this PR
- Restart the `superset_app` container
- Try again creating the chart with the timestamp filter

### ADDITIONAL INFORMATION


- [X] Has associated issue: https://github.com/apache/superset/issues/27382
